### PR TITLE
vscode, vscodium: 1.73.1 -> 1.74.0

### PR DESCRIPTION
**Pull request by eadwu**
###### Description of changes

Has been broken on insiders for probably around a month now. Finally got annoyed enough to spend some time to fix.

- Closes #205042
- Closes #205074
- Closes #205672
- Closes #205661

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/vscode-utils.nix [hunk 1]
```diff
@@ -5,6 +5,8 @@ let
     src,
     # Same as "Unique Identifier" on the extension's web page.
     # For the moment, only serve as unique extension dir.
+    vscodeExtPublisher,
+    vscodeExtName,
```
Comment (linsui) [inline pkgs/applications/editors/vscode/extensions/vscode-utils.nix:9]: These item are not set for some extensions not built with `extensionsFromVscodeMarketplace`. We should either extract them from `vscodeExtUniqueId` or add them to those extensions.

File: pkgs/applications/editors/vscode/with-extensions.nix [hunk 2]
```diff
@@ -46,12 +46,48 @@ let
   wrappedPkgVersion = lib.getVersion vscode;
   wrappedPkgName = lib.removeSuffix "-${wrappedPkgVersion}" vscode.name;
 
+  toExtensionJsonEntry = drv: rec {
+    identifier = {
+      id = "${drv.vscodeExtPublisher}.${drv.vscodeExtName}";
+      uuid = "";
+    };
+
+    version = drv.version;
```
Comment (linsui) [inline pkgs/applications/editors/vscode/with-extensions.nix:55]: This also doesn't always exist.

File: pkgs/applications/editors/vscode/extensions/vscode-utils.nix (hunk 1 above)
Comment (linsui) [reply pkgs/applications/editors/vscode/extensions/vscode-utils.nix:9]: Maybe all extensions should define `mktplcRef`.

File: pkgs/applications/editors/vscode/with-extensions.nix (hunk 2 above)
Comment (linsui) [reply pkgs/applications/editors/vscode/with-extensions.nix:55]: The python extension,

Comment (linsui) [reply pkgs/applications/editors/vscode/with-extensions.nix:55]: Sorry, my mistake.

**Review by linsui — APPROVED** [review]
Works for me. Thanks!

**Comment by bobby285271** [comment]
(Some notes for reviewers)

Ofborg failures are (hopefully) not related to this bump:
https://hydra.nixos.org/job/nixpkgs/trunk/vscode-extensions.vadimcn.vscode-lldb.x86_64-darwin
https://github.com/NixOS/nixpkgs/pull/159220#issuecomment-1036231572

To test some extensions, e.g. `vscode-extensions.vadimcn.vscode-lldb`, use 

```
nix-build -I nixpkgs=. -E 'with import <nixpkgs> { config = { allowUnfree = true; }; }; vscode-with-extensions.override { vscodeExtensions = with vscode-extensions; [ vscode-extensions.vadimcn.vscode-lldb ]; }'
```

**Don't just** `nix-build -A vscode-extensions.vadimcn.vscode-lldb`, e.g. the former will fail to eval on ca331f365ab2045c452f20dba71e54a2b6ee1d59 but the latter does build.

I haven't tested all manually packaged extensions yet.

**Comment by adamcstephens** [comment]
May want to update the title of this PR if it's going to bump the vscode versions.

File: pkgs/applications/editors/vscode/with-extensions.nix (hunk 2 above)
Comment (bobby285271) [reply pkgs/applications/editors/vscode/with-extensions.nix:55]: Yeah I think version should be something required for mktplcRef:

https://github.com/NixOS/nixpkgs/blob/126bb19304148f3b6f76c35167932fbcc32648ee/pkgs/applications/editors/vscode/extensions/vscode-utils.nix#L56

https://github.com/NixOS/nixpkgs/blob/126bb19304148f3b6f76c35167932fbcc32648ee/pkgs/applications/editors/vscode/extensions/python/default.nix#L19-L24

**Comment by schuelermine** [comment]
I’m currently trying to build my home-manager configuration with this PR applied, I will review it then

**Comment by schuelermine** [comment]
Doesn’t work for me. I tried building my home-manager configuration with a nixpkgs set to this PR’s head and VS Code still boots w/o extensions. For reference, my configuration: https://github.com/schuelermine/home-manager-configuration/tree/test/205080 (I had added a `nixpkgs-test` input in flake.nix and an overlay in config/vscode.nix that uses `nixpkgs-test`)

File: pkgs/applications/editors/vscode/with-extensions.nix [hunk 3]
```diff
@@ -46,12 +46,48 @@ let
...
+      targetPlatform = "undefined";
+      isApplicationScoped = false;
+      updated = false;
+      isPreReleaseVersion = false;
+      installedTimestamp = 0;
+      preRelease = false;
+    };
+  };
+
+  extensionJson = builtins.toJSON(map toExtensionJsonEntry vscodeExtensions);
```
Comment (schuelermine) [inline pkgs/applications/editors/vscode/with-extensions.nix:77]: Is there a reason not to leave a space here?

**Review by schuelermine — COMMENTED** [review]
`vscode-with-extensions` works on my machine, but I get an error when the Release Notes page is loaded: “Error loading webview: Error: Could not register service workers: InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state..”

**Comment by bobby285271** [comment]
> InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state.

Sounds like an existing issue, https://github.com/NixOS/nixpkgs/issues/188222

---

Hi @eadwu, looks like you just reverted some previous changes I pushed, if you don't wish this branch to be pushed do you mind fixing both the editorconfig error and the ofborg-eval error, and disable `Allow edits and access to secrets by maintainers` from this page?

**Comment by adamcstephens** [comment]
> Hi @eadwu, looks like you just reverted some previous changes I pushed

Now this PR is extra confusing because it doesn't include the upgrades anymore. :/

**Comment by eadwu** [comment]
Feel free to use this as a base in your local branch.

I usually don't have a lot of time nowadays and don't quite check GitHub on the daily as much (got lucky here - otherwise your work would have been overwritten everyday).

My daily workflow for updates usually involves a `git push -u --all --force` for reasons that probably don't apply anymore.

**Comment by eadwu** [comment]
Note that `vscode-fhs` is likely still broken after this and will require some butchery on the wrapper script since the `extensions.json` would have to be writable to still install extensions outside of Nix.

**Comment by bobby285271** [comment]
Hmm, it that a regression introduced in this update? I think we don't pass `--extensions-dir <nix-store-path>` to vscode-fhs in the wrapper at all and it should be something sitting in `~/.vscode/extensions/extensions.json` :thinking: 

```
# bobby285271 @ inspiron in ~/nixpkgs on git:vscode o [9:00:53] 
$ cat $(nix-build -I nixpkgs=. -E 'with import <nixpkgs> { config = { allowUnfree = true; }; }; vscode-with-extensions.override { vscodeExtensions = with vscode-extensions; [ ]; }')/bin/code
#! /nix/store/nfgpn9av331q7zi1dl6d5qpir60y513s-bash-5.1-p16/bin/bash -e
exec "/nix/store/jjyyy1jdvg9id01mv4y43j76rnr34pii-vscode-1.74.0/bin/code"  --extensions-dir /nix/store/xx6c96p08c9xsq2p5q37msyg7wj4i741-vscode-extensions/share/vscode/extensions "$@" 

# bobby285271 @ inspiron in ~/nixpkgs on git:vscode o [9:01:13] 
$ cat $(nix-build -A vscode-fhs)/bin/code | tail
  --ro-bind-try $(/nix/store/h48pjfgsjl75bm7f3nxcdcrqjkqwns7m-coreutils-9.1/bin/readlink -m /etc/ca-certificates) /etc/ca-certificates
  --ro-bind-try $(/nix/store/h48pjfgsjl75bm7f3nxcdcrqjkqwns7m-coreutils-9.1/bin/readlink -m /etc/pki) /etc/pki
  "${ro_mounts[@]}"
  "${symlinks[@]}"
  "${auto_mounts[@]}"
  
  /nix/store/q75szbm4d0dc0w099inbjdq822rh74vi-code-init/bin/code-init "$@"
)
exec "${cmd[@]}"


# bobby285271 @ inspiron in ~/nixpkgs on git:vscode o [9:02:03] 
$ ls /nix/store/xx6c96p08c9xsq2p5q37msyg7wj4i741-vscode-extensions/share/vscode/extensions   
extensions.json
```

**Pull request closed without merging**